frontend/src/metabase-lib/queries/structured/Aggregation.ts

```diff
@@ -134,7 +137,16 @@ export default class Aggregation extends MBQLClause {
 
   baseType() {
     const short = this.short();
-    return INTEGER_AGGREGATIONS.has(short) ? TYPE.Integer : TYPE.Float;
+    if (INTEGER_AGGREGATIONS.has(short)) {
+      return TYPE.Integer;
+    }
+
+    const field = this.dimension()?.field;
+    if (ORIGINAL_FIELD_TYPE_AGGREGATIONS.has(short) && field) {
+      return field.base_type;
```

Comment:
for `min` and `max` aggregation functions. Uses the base field`s `base_type` instead.

e.g. min(Name) = `type/Text`, min(CreatedAt) = `type/DateTime`, etc.

Reply:
what would happen if we simply check for `field != null` and use its `base_type`? would it work, or is there a case where it doesn't?

Reply:
Yeah, [a unit test fails](https://github.com/metabase/metabase/blob/f71f20e18788f2997a55e8bdafb1a39843d06a13/frontend/test/metabase-lib/lib/Dimension.unit.spec.js#L796-L802) (sorry couldn't find the failed run, I should have done that yesterday, now it's too hard to find). And I think that makes sense, we can't always relying on the `field.base_type`.

Let's say we do **Count Created At** the result field type should be integer still, as stated in the unit test.

Reply:
Makes sense, thanks for the detailed explanation